fvm/evm/types/result.go

```diff
@@ -123,19 +124,25 @@ func (res *Result) VMErrorString() string {
 // can be used by json-rpc and other integration to be returned.
 //
 // This is method is also used to construct block receipt root hash
-// which requires the return receipt satisfy RLP encoding and cover these feilds
+// which requires the return receipt satisfy RLP encoding and cover these fields
 // Type (txType), PostState or Status, CumulativeGasUsed, Logs and Logs Bloom
 // and for each log, Address, Topics, Data (consensus fields)
 // During execution we also do fill in BlockNumber, TxIndex, Index (event index)
 func (res *Result) Receipt() *gethTypes.Receipt {
 	if res.Invalid() {
 		return nil
 	}
+
 	receipt := &gethTypes.Receipt{
-		Type:              res.TxType,
-		CumulativeGasUsed: res.GasConsumed, // TODO: update to capture cumulative
```

Comment:
are we sure about removing the todo, I think this would be an opportunity to actually calculate cumulative gas, which it should sum all the gas used by previous transactions

Reply:
Good point, I think we can update the logic in this PR is we want ?

Reply:
updated at [c6ded6c](https://github.com/onflow/flow-go/pull/6195/commits/c6ded6cd63d8374fc611f778ec3f1e4e0e758889)